pkg/reconcile/pipeline/handler/collect/impl.go

```diff
@@ -131,6 +132,23 @@ func ProvisionedService(ctx pipeline.Context) {
 	}
 }
 
+func DirectSecretReference(ctx pipeline.Context) {
+	services, _ := ctx.Services()
+
+	for _, service := range services {
+		res := service.Resource()
+		if res.GetKind() == "Secret" && res.GetAPIVersion() == "v1" && res.GroupVersionKind().Group == "" {
+			name := res.GetName()
+			secret, err := ctx.ReadSecret(res.GetNamespace(), name)
+			if err != nil {
+				requestRetry(ctx, ErrorReadingSecret, err)
```

Comment:
The request will be retried infinitely in the case the secret does not exist; `err` must be checked and retries shouldn't be requested in unrecoverable cases.

Reply:
What if someone creates a secret after the ServiceBinding resource creation?
I can see we have similar logic in ProvisionedService also. line 111-112.

Reply:
Kubernetes offers mechanisms to react on creation, update and deletion of resources to avoid situations such as the indicated. The appropriated approach would be a) fail without retry if a Secret is not found at the time of the reconciliation and b) observe the life-cycle (either fully or partially) of Secrets to perform the reconciliation of the related ServiceBinding.

Reply:
As I understand we removed all watches as part of optimization. So, I am not sure how to reconcile it.
As I pointed out, this design approach is there in other parts of our codebase.